consumer/consumer.go

```diff
@@ -0,0 +1,187 @@
+package consumer
+
+import (
+	"fmt"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/Shopify/sarama"
+	cluster "github.com/bsm/sarama-cluster"
+	"github.com/heetch/felice/consumer/handler"
+	"github.com/heetch/felice/message"
+	"github.com/pkg/errors"
+)
+
+// Consumer is a Kafka consumer.
+type Consumer struct {
+	consumer      *cluster.Consumer
+	config        *cluster.Config
+	handlers      *handler.Collection
+	wg            sync.WaitGroup
+	quit          chan struct{}
+	RetryInterval time.Duration
+	Metrics       MetricsReporter
+}
+
+// Handle registers the handler for the given topic.
+// Handle must be called before Serve.
+func (c *Consumer) Handle(topic string, h handler.Handler) {
+	c.setup()
+
+	c.handlers.Set(topic, h)
+}
+
+func (c *Consumer) setup() {
+	if c.handlers == nil {
+		c.handlers = &handler.Collection{}
```

Comment:
:tulip: What is the reasoning behind having `c.handlers` as a pointer since all we do is instantiate it like this? Shouldn't we simply use a value field instead of a pointer field?